test/v5/createAmendForeignPropertyAnnualSubmission/def1/Def1_CreateAmendForeignPropertyAnnualSubmissionValidatorSpec.scala

```diff
@@ -0,0 +1,821 @@
+
+        result shouldBe Left(ErrorWrapper(correlationId, BusinessIdFormatError))
+      }
+
+      "passed an empty request body" in {
+        setupMocks()
+        val emptyBody: JsValue = Json.parse("""{}""")
+        val result: Either[ErrorWrapper, CreateAmendForeignPropertyAnnualSubmissionRequestData] =
+          validator(validNino, validBusinessId, validTaxYear, emptyBody).validateAndWrapResult()
+
+        result shouldBe Left(ErrorWrapper(correlationId, RuleIncorrectOrEmptyBodyError))
+      }
+
+      "passed a request body with an empty or missing field" when {
+        List(
+          "/foreignFhlEea",
+          "/foreignFhlEea/allowances"
+        ).foreach(path => testRuleIncorrectOrEmptyBodyWith(bodyWith(entry).replaceWithEmptyObject(path), path))
+
+        testRuleIncorrectOrEmptyBodyWith(
+          bodyWith(entry).replaceWithEmptyObject("/foreignFhlEea/adjustments"),
+          "/foreignFhlEea/adjustments/periodOfGraceAdjustment")
+
+        List(
+          (bodyWith(), "/foreignNonFhlProperty"),
+          (bodyWith(entry.removeProperty("/countryCode")), "/foreignNonFhlProperty/0/countryCode"),
+          (bodyWith(entry.removeProperty("/adjustments").removeProperty("/allowances")), "/foreignNonFhlProperty/0"),
+          (bodyWith(entry.replaceWithEmptyObject("/adjustments")), "/foreignNonFhlProperty/0/adjustments"),
+          (bodyWith(entry.replaceWithEmptyObject("/allowances")), "/foreignNonFhlProperty/0/allowances"),
+          (bodyWith(entryWith("AFG")), "/foreignNonFhlProperty/0/allowances/structuredBuildingAllowance"),
+          (
+            bodyWith(entryWith("AFG", validStructuredBuildingAllowance.removeProperty("/amount"))),
+            "/foreignNonFhlProperty/0/allowances/structuredBuildingAllowance/0/amount"
+          ),
+          (
+            bodyWith(entryWith("AFG", validStructuredBuildingAllowance.removeProperty("/firstYear/qualifyingDate"))),
+            "/foreignNonFhlProperty/0/allowances/structuredBuildingAllowance/0/firstYear/qualifyingDate"
+          ),
+          (
+            bodyWith(entryWith("AFG", validStructuredBuildingAllowance.removeProperty("/firstYear/qualifyingAmountExpenditure"))),
+            "/foreignNonFhlProperty/0/allowances/structuredBuildingAllowance/0/firstYear/qualifyingAmountExpenditure"
+          ),
+          (
+            bodyWith(entryWith("AFG", validStructuredBuildingAllowance.removeProperty("/building/"))),
+            "/foreignNonFhlProperty/0/allowances/structuredBuildingAllowance/0/building"
+          ),
+          (
+            bodyWith(entryWith("AFG", validStructuredBuildingAllowance.removeProperty("/building/postcode"))),
+            "/foreignNonFhlProperty/0/allowances/structuredBuildingAllowance/0/building/postcode"
+          )
+        ).foreach((testRuleIncorrectOrEmptyBodyWith _).tupled)
+      }
+
+      "passed a request body with empty fields except for additional (non-schema) properties" in {
```

Comment:
This test was removed in def2. Can we re-instate it there (but for an unknown field inside the standard foreign property array).